CleverPush/Source/CleverPushInstance.m

```diff
@@ -885,54 +890,58 @@ - (void)subscribe:(CPHandleSubscribedBlock)subscribedBlock failure:(CPFailureBlo
         [center getNotificationSettingsWithCompletionHandler:^(UNNotificationSettings *_Nonnull notificationSettings) {
             if (subscriptionId == nil && channelId != nil && notificationSettings.authorizationStatus == UNAuthorizationStatusNotDetermined) {
                 [self setConfirmAlertShown];
+            } else if (!autoRequestNotificationPermission && notificationSettings.authorizationStatus == UNAuthorizationStatusAuthorized ) {
+                [CleverPush setAutoRequestNotificationPermission:true];
```

Comment:
Why is this required?

Reply:
This is for if the user first sets the `autoRequestNotificationPermission` to false and then the user gives the notification permission access, then we need to set the flag `autoRequestNotificationPermission` to yes because if we don't set the true value, it will go into the false block every where we have to keep the condition like `if (!autoRequestNotificationPermission)`.